### PR TITLE
118 convert yaml files for gridlines interface

### PR DESCRIPTION
**Pull request by lambertaa**
# Reviewer Checklist
https://github.com/NRLMMD-GEOIPS/.github/blob/main/.github/review-template.md

# Related Issues
fixes NRLMMD-GEOIPS/geoips#118

# Testing Instructions
To test if your yaml files fit to the yaml schema applied to them, use this line in a python editor
from geoips.schema import validate
then use the validate function to ensure that each converted yaml file fits to the base schema applied.
(See output for an example)

# Summary

From issue NRLMMD-GEOIPS/geoips#118: 2023-04-04, convert yaml files for gridlines interface
* Create yaml files for gridlines following gridlines schema
created: /geoips/interface_modules/gridlines/default.yaml
created: /geoips/interface_modules/gridlines/tc_0p25degree.yaml
created: /geoips/interface_modules/gridlines/tc_pmw.yaml
created: /geoips/interface_modules/gridlines/tc_visir_3200km.yaml
created: /geoips/interface_modules/gridlines/tc_visir.yaml
created: /geoips/interface_modules/gridlines/tc_windspeed.yaml

# Output
>>> from geoips.schema import validate
>>> from glob import glob
>>> files = glob('geoips/interface_modules/gridlines/*.yaml')
>>> files
['geoips/interface_modules/gridlines/default.yaml', 'geoips/interface_modules/gridlines/tc_0p25degree.yaml', 'geoips/interface_modules/gridlines/tc_pmw.yaml', 'geoips/interface_modules/gridlines/tc_windspeed.yaml', 'geoips/interface_modules/gridlines/tc_visir.yaml', 'geoips/interface_modules/gridlines/tc_visir_3200km.yaml']
>>> for file in files:
...     validate(file)
... 
{'interface': 'gridlines', 'metadata': {'name': 'default', 'family': 'standard', 'short-description': 'The default gridlines configuration', 'long-description': 'Top and left gridline labels, latitude and longitude lines colored\nblack, auto spacing, 1px linewidth, and [4, 2] linestyle\n'}, 'spec': {'labels': {'top': True, 'bottom': False, 'left': True, 'right': False}, 'latitude-lines': {'color': 'black', 'linestyle': [4, 2], 'linewidth': 1, 'spacing': 'auto'}, 'longitude-lines': {'color': 'black', 'linestyle': [4, 2], 'linewidth': 1, 'spacing': 'auto'}}}
{'interface': 'gridlines', 'metadata': {'name': 'tc_0p25degree', 'family': 'standard', 'short-description': 'TC 0.25 degree gridlines configuration', 'long-description': 'All gridline labels enabled, latitude and longitude lines colored\n#00B4B4B4, 0.25 degree spacing, 1px linewidth, and [4, 2] linestyle\n'}, 'spec': {'labels': {'top': True, 'bottom': True, 'left': True, 'right': True}, 'latitude-lines': {'color': '#00B4B4B4', 'linestyle': [4, 2], 'linewidth': 1, 'spacing': 0.25}, 'longitude-lines': {'color': '#00B4B4B4', 'linestyle': [4, 2], 'linewidth': 1, 'spacing': 0.25}}}
{'interface': 'gridlines', 'metadata': {'name': 'tc_pmw', 'family': 'standard', 'short-description': 'TC Passive Microwave gridlines configuration', 'long-description': 'All gridline labels enabled, latitude and longitude lines colored\n#00B4B4B4, 2 degree spacing, 1px linewidth, and [4, 2] linestyle\n'}, 'spec': {'labels': {'top': True, 'bottom': True, 'left': True, 'right': True}, 'latitude-lines': {'color': '#00B4B4B4', 'linestyle': [4, 2], 'linewidth': 1, 'spacing': 2}, 'longitude-lines': {'color': '#00B4B4B4', 'linestyle': [4, 2], 'linewidth': 1, 'spacing': 2}}}
{'interface': 'gridlines', 'metadata': {'name': 'tc_windspeed', 'family': 'standard', 'short-description': 'TC Windspeed gridlines configuration', 'long-description': 'All gridline labels enabled, latitude and longitude lines colored\n#00B4B4B4, 2 degree spacing, 1px linewidth, and [4, 2] linestyle\n'}, 'spec': {'labels': {'top': True, 'bottom': True, 'left': True, 'right': True}, 'latitude-lines': {'color': '#00B4B4B4', 'linestyle': [4, 2], 'linewidth': 1, 'spacing': 2}, 'longitude-lines': {'color': '#00B4B4B4', 'linestyle': [4, 2], 'linewidth': 1, 'spacing': 2}}}
{'interface': 'gridlines', 'metadata': {'name': 'tc_visir', 'family': 'standard', 'short-description': 'TC Visible Infrared gridlines configuration', 'long-description': 'All gridline labels enabled, latitude and longitude lines colored\n#000000, 2 degree spacing, 1px linewidth, and [4, 2] linestyle\n'}, 'spec': {'labels': {'top': True, 'bottom': True, 'left': True, 'right': True}, 'latitude-lines': {'color': '#000000', 'linestyle': [4, 2], 'linewidth': 1, 'spacing': 2}, 'longitude-lines': {'color': '#000000', 'linestyle': [4, 2], 'linewidth': 1, 'spacing': 2}}}
{'interface': 'gridlines', 'metadata': {'name': 'tc_visir_3200km', 'family': 'standard', 'short-description': 'TC Visible Infrared 3200km gridlines configuration', 'long-description': 'All gridline labels enabled, latitude and longitude lines colored\n#000000, 5 degree spacing, 1px linewidth, and [4, 2] linestyle\n'}, 'spec': {'labels': {'top': True, 'bottom': True, 'left': True, 'right': True}, 'latitude-lines': {'color': '#000000', 'linestyle': [4, 2], 'linewidth': 1, 'spacing': 5}, 'longitude-lines': {'color': '#000000', 'linestyle': [4, 2], 'linewidth': 1, 'spacing': 5}}}

### CODE REVIEW

**Comment by lance-wilson** [comment]
Note that this also includes the "bad" schema files that I added.

**Comment by jsolbrig** [comment]
Thanks @lance-wilson @lambertaa, this all looks good. I will approve this as soon as someone is able to approve #135 for me. I've asked @mindyls to take a look at that.

File: geoips/schema/gridlines.yaml
Comment (mindyls) [inline geoips/schema/gridlines.yaml:80]: @jsolbrig Note that the "grid_lat_xoffset", etc fields would be used to specify the offset for the lat/lon labels - it is not implemented right now, but I think it is probably something that we will want to support.

**Review by mindyls — APPROVED** [review]
This looks good to me, and I tested via #135 .

Read and close my conversation (adding label positioning support can be a separate Issue, but just wanted to make a note of it).  Good to merge after #135 is merged.

File: docs/source/releases/v1_10_0_dev0.rst
Comment (mindyls) [inline docs/source/releases/v1_10_0_dev0.rst:4]: Note I added bulleted summary list right under "Version" header, and made the "From issue" lines bold rather than a bullet.

File: geoips/schema/gridlines.yaml
Comment (jsolbrig) [reply geoips/schema/gridlines.yaml:80]: @mindyls Should we just add that in when we add support for it?

File: docs/source/releases/v1_10_0_dev0.rst
Comment (jsolbrig) [reply docs/source/releases/v1_10_0_dev0.rst:4]: Got it. I will try to do this from now on.

File: geoips/schema/gridlines.yaml
Comment (jsolbrig) [reply geoips/schema/gridlines.yaml:80]: See #147

File: docs/source/releases/v1_10_0_dev0.rst
Comment (jsolbrig) [reply docs/source/releases/v1_10_0_dev0.rst]: Okay, understood.

**Pull request merged**
